internal/enginenetx/statsmanager.go

```diff
@@ -87,6 +88,86 @@ type statsTactic struct {
 	Tactic *httpsDialerTactic
 }
 
+// statsNilSafeSuccessRate is a convenience function for computing the success rate
+// which returns zero as the success rate if CountStarted is zero
+//
+// for robustness, be paranoid about nils here because the stats are
+// written on the disk and a user could potentially edit them
```

Comment:
```suggestion
// For robustness, be paranoid about nils here because the stats are
// written on the disk and a user could potentially edit them.
```